management-api-agent-common/src/main/java/io/k8ssandra/metrics/http/NettyHttpInitializer.java

```diff
@@ -0,0 +1,28 @@
+package io.k8ssandra.metrics.http;
+
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.codec.http.HttpContentCompressor;
+import io.netty.handler.codec.http.HttpServerCodec;
+import io.netty.handler.ssl.SslContext;
+
+public class NettyHttpInitializer extends ChannelInitializer<SocketChannel> {
+    private final SslContext sslCtx;
+
+    public NettyHttpInitializer(SslContext sslCtx) {
+        this.sslCtx = sslCtx;
+    }
+
+    @Override
+    public void initChannel(SocketChannel ch) {
+        ChannelPipeline p = ch.pipeline();
+        if (sslCtx != null) {
+            p.addLast(sslCtx.newHandler(ch.alloc()));
+        }
+        p.addLast(new HttpServerCodec());
+        // TODO Do we need chunking for larger /metrics ?
```

Comment:
I think we should account for clusters with 100s of tables and implement whatever we can to make this work for these as well.
How hard is chunking to implement?
We could probably create a follow up ticket for this.

Reply:
Chunking isn't, but I'm not sure if Prometheus scraper understands it. I don't think the 100s of tables is an issue here, as long as the reading client can handle it. 

We anyway process it all in the same moment.